File: leetcode/128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.py
# Last updated: 29/04/2026, 23:12:26
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        res = 0

        for n in nums:
            if (n-1) not in seen:
                cnt = 0
                while (n+cnt) in seen:
                    cnt += 1
                res = max(res,cnt)
        return res

File: leetcode/128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.cpp
// Last updated: 29/04/2026, 23:12:31
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> uniqueNums;
        for (int i = 0; i< nums.size(); i++) {
            uniqueNums.insert(nums[i]);
        }

        int maxSeq = 0;
        int currentSeq = 0;
        for (int i = 0; i< nums.size(); i++) {
            int j = 0;
            if (uniqueNums.find(nums[i] - 1) == uniqueNums.end()) {
                while (uniqueNums.find(nums[i] + j) != uniqueNums.end()) {
                    j++;
                    currentSeq++;
                }
                maxSeq = max(maxSeq, currentSeq);
                currentSeq = 0;
            }
        }

        return maxSeq;
    }
};